pkg/apis/serving/v1alpha2/inferenceservice_types.go

```diff
@@ -60,6 +60,29 @@ type DeploymentSpec struct {
 	// This is the up bound for autoscaler to scale to
 	// +optional
 	MaxReplicas int `json:"maxReplicas,omitempty"`
+	// Activate request/response logging
+	// +optional
+	InferenceLogger *InferenceLogger `json:"inferenceLogger,omitempty"`
+}
+
+type InferenceLoggerType string
+
+const (
+	InferenceLogBoth     InferenceLoggerType = "both"
+	InferenceLogRequest  InferenceLoggerType = "request"
+	InferenceLogresponse InferenceLoggerType = "response"
+)
+
+type InferenceLogger struct {
+	// URL to send request logging CloudEvents
+	// +optional
+	Url *string `json:"url,omitempty"`
+	// What payloads to log
+	// +optional
+	LogType *InferenceLoggerType `json:"logType,omitempty"`
```

Comment:
redundant "log".

What about "type" or "mode"?

Reply:
We should probably make this required.

Reply:
Mode sounds good. Was assuming if not there implies both which will probably be most common. So wanted minimize unnecessary yaml.

Reply:
We just never want a case where we have `logger: {}`

Reply:
Well if url is not specified or mode (both pointers) then yes you can have `logger: {}`. Is this an anti-pattern in k8s?

Reply:
It's not something I commonly see, though I can't speak authoritatively. Semantically it means it exists, but it can seem "empty"